### DISCUSSION TITLE
How can I cache images?

### DISCUSSION THREAD

**Discussion by kalvn** [mpdf/mpdf#1462] [Q&A] [answered]
Hello there,

I'm wondering if there's a way to cache external images referenced in the HTML when rendering the PDF so that it's not fetched all the time when I run the generation several times in a row.

**Comment by finwe** [comment, marked as answer]
Currently the only way is to implement a caching proxy outside of mPDF. In the future versions, there can be a PSR-7/PSR-18 implementation of an internal HTTP client supporting middleware with the ability to cache.

**Reply by kalvn** [reply]
Alright, thanks for the reply.
